libraries/chain/block_log.cpp

```diff
@@ -84,24 +97,34 @@ namespace eosio { namespace chain {
             template <typename ChainContext, typename Lambda>
             static fc::optional<ChainContext> extract_chain_context( const fc::path& data_dir, Lambda&& lambda );
 
-            static std::vector<char> pack(const pruned_block&                     b,
-                                          pruned_transaction::cf_compression_type segment_compression);
-            static void unpack(fc::cfile_datastream& ds, log_entry& entry);
+            static std::vector<char> pack_v4_log_entry(const pruned_block&                     b,
+                                                       pruned_transaction::cf_compression_type segment_compression);
 
             uint64_t write_log_entry(const pruned_block&                     b,
                                      pruned_transaction::cf_compression_type segment_compression);
 
             void             read_block_header(block_header& bh, uint64_t file_pos);
             pruned_block_ptr read_block(uint64_t pos);
             void             read_head();
-            log_entry        read_v4_log_entry(uint64_t pos);
+            void             read_v4_log_entry(uint64_t pos, log_entry&);
```

Comment:
I mad the change because all but one unpack log entry usage needs the heap allocated pruned_block. Most of the usage use it in a loop and discard the value every iteration. I consider this is private to block_log.cpp, so the efficiency took precedence.